app/services/ExactTargetService.scala

```diff
@@ -83,14 +86,32 @@ trait ExactTargetService extends LazyLogging {
           promotionDescription = promotionDescription
         )
       )
+    }
+
+    for {
+      sub <- getZuoraPaidSubscription()
```

Comment:
Now you're using functions to return futures these steps won't happen concurrently, before that the futures were begun outside of the for comprehension